### ISSUE TITLE
Camera.getPose() is not oriented to physical camera

### ISSUE THREAD

**Issue by deicide199** [google-ar/arcore-android-sdk#535]
Doing some rudimentary testing with the display transform returned by the frame and the pose returned by the camera, I've found that when the display transform indicated no rotation (i.e. the screen and the physical camera are aligned) the device was in a landscape orientation. Inspecting the Camera.getPose() documentation, it seems it should also be aligned with this device orientation -- however the Camera.getPose() matrix is actually aligned with the _portrait_ orientation, not matching the documentation.

I've verified this on a Pixel 2 and a few Samsung phones on the latest ArCore.

Thanks!

**Comment by inio** [comment]
`Camera.getPose()` should be located at the camera and oriented such that local -Z points in the direction that the camera is looking, the first pixel of the CPU image is looking in the local -X+Y-Z direction, and the last pixel of the first row is in the local +X+Y-Z direction.

`Camera.getDisplayOrientedPose()` should be located at the camera and oriented so that the +Y direction is approximately up on the logical screen.  It will always differ from `getPose()` by exactly a multiple of 90° rotations about Z.

`Frame.getAndroidSensorPose()` should be located at the IMU and oriented with the android sensor coordinate system. For example, the gravity sensor readout when translated by this pose should point approximately +Y.

**Comment by deicide199** [comment]
Ah, so for that first one, `Camera.getPose()` is not what I have observed. In order to find out which is the first pixel of the CPU image, I use the display transform. When there is no rotation in the display transform, that means the first pixel of the image will align with the first pixel of my screen, correct? The `Pose` from `Camera.getPose()` returned does not align with that "orientation" though -- meaning to say, `Camera.getPose() != Camera.getDisplayOrientedPose()` in this orientation -- which should be the case according to the documentation:
> The position and orientation of the pose follow the device's physical camera

Although reading that, I suppose it's possible that _is_ correct and the hardware sensors are all in portrait and you guys rotate the image manually so that the image is given horizontal and the display transform is aligned with that? 

My fix right now it to manually apply a rotation to the `Camera.getPose()` to align it with the image/hardware sensor so my math works out. I'm just trying to get some insight as to whether this will be fixed/changed at some point -- as my code will need to change, then 😬

**Comment by inio** [comment]
This is starting to sound more like a bug.  You said this was on a Pixel 2 and various Samsung phones?

**Comment by deicide199** [comment]
Yeah. Pixel 2, Samsung S8, and S7 for sure.

**Comment by inio** [comment]
We've confirmed this bug.  Oops!  We'll try to fix this in a future version in as backwards-compatible way as possible.

**Comment by inio** [comment]
Due to timing this is going to take a bit longer than I'd like to fix properly.

As a workaround, I can confirm that it is "correct" to compose the pose returned by `getPose()` with a 90° rotation about the Z axis (though I'm not sure not sure which direction) and you will get the correct physical pose. 
 That is:

```
static final Pose CAMERA_POSE_FIX = 
    Pose.makeRotation(0, 0, ±Math.sqrt(0.5f), Math.sqrt(0.5f)); // ±90° rotation about Z
...
Pose physicalPose = camera.getPose().compose(CAMERA_POSE_FIX);
```

The above code should work properly for all current and future versions of ARCore on all devices.  This issue will be fixed in a future version by deprecating `Camera.getPose()` and introducing a new API that returns the correct pose.

**Comment by inio** [comment]
Update: On one device, the Nexus 5X, the above is incorrect.  The following code should work on ALL devices:

    static Pose deviceToPhysical;
    
    // Can be done once after camera permission is granted
    CameraManager cm = getSystemService(CameraManager.class);
    int sensorOrientation = Arrays.stream(cm.getCameraIdList()).map((id) -> {
      try {
        if (cm.getCameraCharacteristics(id).get(CameraCharacteristics.LENS_FACING) ==
                CameraMetadata.LENS_FACING_BACK) {
          return cm.getCameraCharacteristics(id).get(CameraCharacteristics.SENSOR_ORIENTATION);
        }
      } catch (CameraAccessException e) {
      }
      return -1;
    }).filter((orientation) -> orientation != -1).findFirst().orElse(0);
    deviceToPhysical = Pose.makeInterpolated(
            Pose.IDENTITY,
            Pose.makeRotation(0, 0, -(float)Math.sqrt(0.5), (float)Math.sqrt(0.5)),
            sensorOrientation / 90);
   
    // Per frame
    Pose physicalCameraPose = frame.getCamera().getPose().compose(deviceToPhysical);

**Comment by inio** [comment]
This is fixed in ARCore 1.6 which was released yesterday.
